debug math_domain_error

a2not opened the pull request:
## issue
https://github.com/Khdbble/thesis/issues/1

a2not committed:
added try;except branching for debugging

plokiy commented:
```
ValueError at:
- 0.5 * math.log(np.linalg.det(mH)) - 0.5 * np.inner(mU[t], np.linalg.solve(mH, mU[t].T))
mH:
[[6.55204531e+15 6.55204531e+15 6.55204531e+15 6.55204531e+15
  6.55204531e+15 6.55204531e+15 6.55204531e+15]
 [6.55204531e+15 6.55204531e+15 6.55204531e+15 6.55204531e+15
  6.55204531e+15 6.55204531e+15 6.55204531e+15]
 [6.55204531e+15 6.55204531e+15 6.55204531e+15 6.55204531e+15
  6.55204531e+15 6.55204531e+15 6.55204531e+15]
 [6.55204531e+15 6.55204531e+15 6.55204531e+15 6.55204531e+15
  6.55204531e+15 6.55204531e+15 6.55204531e+15]
 [6.55204531e+15 6.55204531e+15 6.55204531e+15 6.55204531e+15
  6.55204531e+15 6.55204531e+15 6.55204531e+15]
 [6.55204531e+15 6.55204531e+15 6.55204531e+15 6.55204531e+15
  6.55204531e+15 6.55204531e+15 6.55204531e+15]
 [6.55204531e+15 6.55204531e+15 6.55204531e+15 6.55204531e+15
  6.55204531e+15 6.55204531e+15 6.55204531e+15]]
mU[t];
[-0.33948541  0.07649587 -1.54032986  2.53612197  2.37920411  1.38012481
  0.16380299]
```
```
ValueError at:
- 0.5 * math.log(np.linalg.det(mH)) - 0.5 * np.inner(mU[t], np.linalg.solve(mH, mU[t].T))
mH:
[[1.26644552e+16 1.26644552e+16 1.26644552e+16 1.26644552e+16
  1.26644552e+16 1.26644552e+16 1.26644552e+16]
 [1.26644552e+16 1.26644552e+16 1.26644552e+16 1.26644552e+16
  1.26644552e+16 1.26644552e+16 1.26644552e+16]
 [1.26644552e+16 1.26644552e+16 1.26644552e+16 1.26644552e+16
  1.26644552e+16 1.26644552e+16 1.26644552e+16]
 [1.26644552e+16 1.26644552e+16 1.26644552e+16 1.26644552e+16
  1.26644552e+16 1.26644552e+16 1.26644552e+16]
 [1.26644552e+16 1.26644552e+16 1.26644552e+16 1.26644552e+16
  1.26644552e+16 1.26644552e+16 1.26644552e+16]
 [1.26644552e+16 1.26644552e+16 1.26644552e+16 1.26644552e+16
  1.26644552e+16 1.26644552e+16 1.26644552e+16]
 [1.26644552e+16 1.26644552e+16 1.26644552e+16 1.26644552e+16
  1.26644552e+16 1.26644552e+16 1.26644552e+16]]
mU[t];
[-0.49028838  2.00843939  1.98265139 -2.49877222 -0.24580449  2.31705868
  1.16801331]
```
```
ValueError at:
- 0.5 * math.log(np.linalg.det(mH)) - 0.5 * np.inner(mU[t], np.linalg.solve(mH, mU[t].T))
mH:
[[6.55204535e+15 6.55204535e+15 6.55204535e+15 6.55204535e+15
  6.55204535e+15 6.55204535e+15 6.55204535e+15]
 [6.55204535e+15 6.55204535e+15 6.55204535e+15 6.55204535e+15
  6.55204535e+15 6.55204535e+15 6.55204535e+15]
 [6.55204535e+15 6.55204535e+15 6.55204535e+15 6.55204535e+15
  6.55204535e+15 6.55204535e+15 6.55204535e+15]
 [6.55204535e+15 6.55204535e+15 6.55204535e+15 6.55204535e+15
  6.55204535e+15 6.55204535e+15 6.55204535e+15]
 [6.55204535e+15 6.55204535e+15 6.55204535e+15 6.55204535e+15
  6.55204535e+15 6.55204535e+15 6.55204535e+15]
 [6.55204535e+15 6.55204535e+15 6.55204535e+15 6.55204535e+15
  6.55204535e+15 6.55204535e+15 6.55204535e+15]
 [6.55204535e+15 6.55204535e+15 6.55204535e+15 6.55204535e+15
  6.55204535e+15 6.55204535e+15 6.55204535e+15]]
mU[t];
[-0.33948541  0.07649587 -1.54032986  2.53612197  2.37920411  1.38012481
  0.16380299]
```
```
ValueError at:
- 0.5 * math.log(np.linalg.det(mH)) - 0.5 * np.inner(mU[t], np.linalg.solve(mH, mU[t].T))
mH:
[[6.55207367e+15 6.55207367e+15 6.55207367e+15 6.55207367e+15
  6.55207367e+15 6.55207367e+15 6.55207367e+15]
 [6.55207367e+15 6.55207367e+15 6.55207367e+15 6.55207367e+15
  6.55207367e+15 6.55207367e+15 6.55207367e+15]
 [6.55207367e+15 6.55207367e+15 6.55207367e+15 6.55207367e+15
  6.55207367e+15 6.55207367e+15 6.55207367e+15]
 [6.55207367e+15 6.55207367e+15 6.55207367e+15 6.55207367e+15
  6.55207367e+15 6.55207367e+15 6.55207367e+15]
 [6.55207367e+15 6.55207367e+15 6.55207367e+15 6.55207367e+15
  6.55207367e+15 6.55207367e+15 6.55207367e+15]
 [6.55207367e+15 6.55207367e+15 6.55207367e+15 6.55207367e+15
  6.55207367e+15 6.55207367e+15 6.55207367e+15]
 [6.55207367e+15 6.55207367e+15 6.55207367e+15 6.55207367e+15
  6.55207367e+15 6.55207367e+15 6.55207367e+15]]
mU[t];
[-0.33948541  0.07649587 -1.54032986  2.53612197  2.37920411  1.38012481
  0.16380299]
```

a2not committed:
dealt with the case where det(mH) = 0

a2not committed:
erased bug reports that are solved

a2not committed:
added debug print

a2not committed:
restored num of reps

a2not committed:
added constraints to optimization process

Assumption 1 & 4 requires some linear constraints to run properly.
These are condition on the parameters rho, gam, varphi, and eta.

plokiy commented:
## debug
Turns out we need some constraints to satisfy the stationary condition.
Regarding Assumptions 1 and 4, pass their linear constraints to the `scipy.optimize.minimize()` at https://github.com/plokiy/panel_garch/pull/2/commits/fe93f2fe9e5292bd376ed513d265adf4ed588314

a2not committed:
refactored
